clients/pkg/promtail/targets/syslog/syslogtarget.go

```diff
@@ -91,15 +94,58 @@ func (t *SyslogTarget) run() error {
 	if err != nil {
 		return fmt.Errorf("error setting up syslog target %w", err)
 	}
+
+	tlsEnabled := len(t.config.TLSConfig.CertFile) > 0 || len(t.config.TLSConfig.CertFile) > 0 || len(t.config.TLSConfig.CAFile) > 0
+	if tlsEnabled {
+		tlsEnabled = true
+		tlsConfig, err := newTLSConfig(t.config.TLSConfig.CertFile, t.config.TLSConfig.KeyFile, t.config.TLSConfig.CAFile)
+		if err != nil {
+			return fmt.Errorf("error setting up syslog target %w", err)
+		}
+		l = tls.NewListener(l, tlsConfig)
+	}
+
 	t.listener = l
-	level.Info(t.logger).Log("msg", "syslog listening on address", "address", t.ListenAddress().String())
+	level.Info(t.logger).Log("msg", "syslog listening on address", "address", t.ListenAddress().String(), "tls", tlsEnabled)
 
 	t.openConnections.Add(1)
 	go t.acceptConnections()
 
 	return nil
 }
 
+func newTLSConfig(certFile string, keyFile string, caFile string) (*tls.Config, error) {
```

Comment:
This is inspired by [NewTLSConfig from github.com/prometheus/common/config](https://pkg.go.dev/github.com/prometheus/common/config#NewTLSConfig).